apps/react/src/features/genres/components/GenreList/GenreList.tsx

```diff
@@ -0,0 +1,57 @@
+import { memo, FC, useState } from 'react';
+import { AnimeGenre } from '@js-camp/core/models/genre.model';
+import { Box, List, ListItem, ListItemButton, ListItemText, IconButton } from '@mui/material';
+import AddIcon from '@mui/icons-material/Add';
+import { useParams } from 'react-router-dom';
+
+import { GenreListItem } from '../GenreListItem';
+import { GenreFilters } from '../GenreFilters';
+
+import styles from './GenreList.module.css';
+
+type GenresListProps = {
+
+	/** An array of anime genres. */
+	readonly genres: Array<AnimeGenre>;
+
+	/** Click handler. */
+	onGenreClick: (id: number) => void;
+};
+
+/** A list of anime genres. */
+const GenresListComponent: FC<GenresListProps> = ({ genres, onGenreClick }: GenresListProps) => {
+	const { genreId } = useParams<{ genreId: string; }>();
+	const [selectedGenreId, setSelectedGenreId] = useState<number | undefined>(genreId ? +genreId : undefined);
```

Comment:
```suggestion
	const [selectedGenreId, setSelectedGenreId] = useState<number | undefined>(genreId ? Number(genreId) : undefined);
```
https://wiki.saritasa.rocks/frontend/languages/ts-js/best-practices#dont-use-implicit-type-casting